### ISSUE TITLE
Custom Dataset Hub Issue

### ISSUE THREAD

**Issue by swjacks** [ultralytics/hub#454]
### Search before asking

- [X] I have searched the HUB [issues](https://github.com/ultralytics/hub/issues) and found no similar bug report.

### HUB Component

Datasets

### Bug

I'm trying to train yolov8 on a custom dataset via the hub.  The images are 1920 x 1200 and the segmented files are in the correct format.  I get the following error when running:

<img width="1142" alt="image" src="https://github.com/ultralytics/hub/assets/53322185/c1914467-f878-48ed-9f2a-e66e84551b99">

This is the structure: 

<img width="560" alt="image" src="https://github.com/ultralytics/hub/assets/53322185/83facaaf-dad0-4e6b-831d-4614bc66240a">

I've spent the last hour or so reading through other issues and cannot seem to solve the problem.  I'm uncertain if this is user error or a bug.  

The segmentation files themselves are in the correct format and in the hub itself the segmentations are shown as masks over the images themselves.  

I've tried deleting the caches and remaking them with no success.  

Additionally I get the following error after 54 epochs: 

<img width="964" alt="image" src="https://github.com/ultralytics/hub/assets/53322185/bbd2784a-5a9c-486e-b88f-698f44aa7cbc">

I suspect this has something to do with the built in dataloader breaking down the 24 training set images into 54 images and something getting borked; however, I have no idea what that could be.  Any advice here would be appreciated! 

Sorry if this is the wrong spot to put this/this ends up not being a bug fix.  

### Environment

Google collab 

### Minimal Reproducible Example

1.) Create new Yolov8 Model via hub
2.) Run model.train()

**Comment by UltralyticsAssistant** [comment]
@swjacks thank you for the detailed report. It looks like you're experiencing two separate issues.

Firstly, let's address the error you're encountering during training. It seems to be related to the aspect ratios of your images and the custom dataset you are using. The dataloader in YOLO attempts to form mini-batches of images and applies size augmentation, which may not handle certain image aspect ratios well. This is not specifically a failure in the hub, but more a characteristic of the method YOLO uses to aggregate images for training.

The second issue appears after 54 epochs with an IndexError. This could be due to an issue with your dataset rather than the hub itself. It might be that one of the images in the dataset doesn't have an associated label, or the label file is not in the correct format. Also, remember that in the YOLO format, all values of bounding box coordinates are relative values, not pixel values, and must be within 0 and 1. Ensure to check your datasets to see if all labels are correctly aligned with the respective images and are in the right format.

As for the dataloader breaking down the 24 training set images into 54 images, that is a standard behavior in YOLO where the dataloader applies certain data augmentations during training, including flipping and scaling of the images.

Please review your dataset files and labels closely to verify they are correctly formatted. If you believe your dataset is correctly formatted and these errors persist, it is possible there may be a bug. In that case, kindly provide all the necessary details (including any code used to prepare the dataset, the total number of images and labels, and a few examples of your label files) to help isolate and diagnose the issue. 

Keep in mind that this assistant does not provide real-time support and may not always provide a complete solution. If you continue to encounter issues, you may want to seek help directly from the community through the Ultralytics Google Group or the GitHub issues page.
